pkg/libvmi/storage.go

```diff
@@ -36,7 +36,23 @@ const (
 func WithContainerDisk(diskName, imageName string) Option {
 	return func(vmi *v1.VirtualMachineInstance) {
 		addDisk(vmi, newDisk(diskName, v1.DiskBusVirtio))
-		addVolume(vmi, newContainerVolume(diskName, imageName))
+		addVolume(vmi, newContainerVolume(diskName, imageName, ""))
+	}
+}
+
+// WithContainerSATADisk specifies the disk name and the name of the container image to be used.
+func WithContainerSATADisk(diskName, imageName string) Option {
+	return func(vmi *v1.VirtualMachineInstance) {
+		addDisk(vmi, newDisk(diskName, v1.DiskBusSATA))
+		addVolume(vmi, newContainerVolume(diskName, imageName, ""))
+	}
+}
+
+// WithContainerDiskAndPath specifies the disk name and the name of the container image, and the path to be used.
+func WithContainerDiskAndPath(diskName, imageName, path string) Option {
```

Comment:
I wander why not to add an additional argument to the existing `WithContainerDisk`.
Callers can just set it with an explicit empty path.

I guess the cost is changing the existing callers.
Have you considered the options?

Reply:
>I guess the cost is changing the existing callers.

Exactly

> Have you considered the options?

yes

Reply:
I think it will be better to just change all callers so we will have a nice API.
But I will not insist or block on this.

Reply:
Agree. I will do it in a follow-up PR.